### PR TITLE
fix: update validation response statusCode to 422 and message to ApiErrors

### PR DESCRIPTION
**Pull request by preethivenkatesh-sdase**

### CODE REVIEW

File: sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/security/handler/ObscuringErrorHandlerTest.java [hunk 1]
```diff
@@ -133,6 +133,36 @@ void shouldGetResource() {
...
+    var headers = new HttpHeaders();
+    headers.setContentType(MediaType.APPLICATION_JSON);
+    var exchange =
+        client.postForEntity(
+            getServerBaseUrl() + "/api/validate",
+            new TestResource().setValue("test").setPostcode("ab123"),
+            ApiError.class);
+    assertThat(exchange)
+        .isNotNull()
+        .extracting(ResponseEntity::getStatusCode, r -> r.getBody().getTitle())
```
Comment (elbiocaetano) [inline sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/security/handler/ObscuringErrorHandlerTest.java:162]: Can we validate the invalidParams field, as well, to check if the response contain the details of the errors?

File: sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/security/handler/ObscuringErrorHandlerTest.java [hunk 2]
```diff
@@ -133,6 +133,36 @@ void shouldGetResource() {
...
+    var headers = new HttpHeaders();
+    headers.setContentType(MediaType.APPLICATION_JSON);
+    var exchange =
+        client.postForEntity(
+            getServerBaseUrl() + "/api/validate",
+            new TestResource().setPostcode("1234"),
+            ApiError.class);
+    assertThat(exchange)
+        .isNotNull()
+        .extracting(ResponseEntity::getStatusCode, r -> r.getBody().getTitle())
```
Comment (elbiocaetano) [inline sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/security/handler/ObscuringErrorHandlerTest.java:147]: Can we validate the invalidParams field, as well, to check if the response contain the details of the errors?

File: sda-commons-web-autoconfigure/src/main/java/org/sdase/commons/spring/boot/web/security/handler/MethodArgumentExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      allNormalized = newNormalized.equals(normalizedToMatchGuava);
+      normalizedToMatchGuava = newNormalized;
+    }
+    // end of backward compatibility implementation to match Guava transformation
+    return ERROR_CODE_TRANSLATOR.translate(normalizedToMatchGuava);
+  }
+
+  private static String normalizeToMatchGuava(String normalizedToMatchGuava) {
+    return normalizedToMatchGuava.replaceAll("([A-Z])([A-Z])", "$1_$2");
+  }
```
Comment (JoergSiebahn) [inline sda-commons-web-autoconfigure/src/main/java/org/sdase/commons/spring/boot/web/security/handler/MethodArgumentExceptionHandler.java:68]: Imho, we don't need to align with the old Guava rules here. We [changed that behavior](https://github.com/SDA-SE/sda-dropwizard-commons/commit/9c5bc8d04eda0061f5a613b6676ebf2705c351aa) for v3 in DW commons.

Comment (preethivenkatesh-sdase) [reply sda-commons-web-autoconfigure/src/main/java/org/sdase/commons/spring/boot/web/security/handler/MethodArgumentExceptionHandler.java]: removed the Guava rules and aligned with new v3 DW commons

**Review by elbiocaetano — APPROVED** [review]

**Review by elbiocaetano — APPROVED** [review]

**Pull request merged**